pages/matches/[id].js

```diff
@@ -48,17 +48,26 @@ export default function MatchDetails({ matches, bars, onDeleteBarOrMatch }) {
       const updatedSelectedBar = { ...selectedBar };
       updatedSelectedBar.matches.push(parseInt(currentMatch.id));
 
-      setUpdatedCurrentBars((prevBars) => [...prevBars, updatedSelectedBar]);
+      const newUpdatedSelectedBar = {
+        ...updatedSelectedBar,
+        showsMatch: updatedSelectedBar.matches?.length > 0 ? true : false,
+      };
+      setUpdatedCurrentBars((prevBars) => [...prevBars, newUpdatedSelectedBar]);
     }
   }
 
   function handleDeleteBar(id) {
-    const updatedBars = bars.map((bar) => {
+    const copyOfBars = [...bars];
+    const updatedBars = copyOfBars.map((bar) => {
```

Comment:
```suggestion
    const updatedBars = bars.map((bar) => {
```
You had it like this before, doesn't it work like that? I think you don't need the `copyOfBars` array because modifying the objects within the bars array directly does not change the reference to those objects. So we can use the bars array directly without creating a separate copy.

Reply:
you're right, changed it :-)